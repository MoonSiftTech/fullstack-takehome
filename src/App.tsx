import { ThemeProvider } from "styled-components";
import { BrowserRouter, Routes, Route } from "react-router";
import { Collection } from "src/components/Collection";
import { batmanTheme } from "src/components/ui/theme";
import "src/App.css";

function App() {
  return (
    <ThemeProvider theme={batmanTheme}>
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Collection />} />
        </Routes>
      </BrowserRouter>
    </ThemeProvider>
  );
}

export default App;
