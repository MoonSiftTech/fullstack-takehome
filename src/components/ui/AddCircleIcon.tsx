import React from "react";

interface AddCircleIconProps {
  className?: string;
  style?: React.CSSProperties;
  size?: number;
  color?: string;
}

export const AddCircleIcon: React.FC<AddCircleIconProps> = ({
  className,
  style,
  size = 24,
  color = "currentColor",
}) => {
  return (
    <svg
      data-testid="add-circle-icon"
      xmlns="http://www.w3.org/2000/svg"
      width={size}
      height={size}
      viewBox="0 0 24 24"
      fill="none"
      stroke={color}
      className={className}
      style={style}
    >
      <circle cx="12" cy="12" r="10" strokeWidth="1" />
      <line x1="12" y1="8" x2="12" y2="16" strokeWidth="1" />
      <line x1="8" y1="12" x2="16" y2="12" strokeWidth="1" />
    </svg>
  );
};
