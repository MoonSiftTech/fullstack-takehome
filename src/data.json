{
  "name": "My Collection",
  "description": "This is my collection for clothes",
  "collectionId": "123",
  "collectionUserId": "123456",
  "username": "Joe Blogs",
  "views": 36,
  "products": [
    {
      "id": "1",
      "price": "From £193.50",
      "savedPrice": "From £193.50",
      "description": "The North Face Saikuru puffer jacket in black | ASOS",
      "url": "https://www.asos.com/the-north-face/the-north-face-saikuru-puffer-jacket-in-black/prd/205418444#colourWayId-205418450",
      "image": "/static/puffer.jpg",
      "pageURL": "www.asos.com/the-north-face/the-north-face-saikuru-puffer-jacket-in-black/prd/205418444",
      "quantity": 4
    },
    {
      "id": "2",
      "price": "£100.00",
      "savedPrice": "£100.00",
      "description": "New Balance New balance 574 trainers in red",
      "url": "https://www.asos.com/new-balance/new-balance-new-balance-574-trainers-in-red/prd/207030180#colourWayId-207030181",
      "image": "/static/sneaker.jpg",
      "pageURL": "www.asos.com/new-balance/new-balance-new-balance-574-trainers-in-red/prd/207030180",
      "quantity": 1
    },
    {
      "id": "3",
      "price": "£40.00",
      "savedPrice": "£40.00",
      "description": "adidas workout Quarter-Zip Track Top in grey",
      "url": "https://www.asos.com/adidas-performance/adidas-workout-quarter-zip-track-top-in-grey/prd/205734439#colourWayId-205734503",
      "image": "/static/workout_top.jpg",
      "pageURL": "www.asos.com/adidas-performance/adidas-workout-quarter-zip-track-top-in-grey/prd/205734439",
      "quantity": 1
    }
  ]
}
