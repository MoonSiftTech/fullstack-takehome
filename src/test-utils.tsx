import React, { ComponentType } from "react";
import { ThemeProvider } from "styled-components";
import {
  render,
  type RenderOptions,
  type RenderResult,
} from "@testing-library/react";

import { batmanTheme } from "src/components/ui/theme";

const Providers = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider theme={batmanTheme}>{children}</ThemeProvider>;
};

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult => {
  return render(ui, { wrapper: Providers as ComponentType, ...options });
};

export * from "@testing-library/react";
export { customRender as render };
