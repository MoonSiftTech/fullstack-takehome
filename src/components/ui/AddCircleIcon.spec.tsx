import { describe, it, expect } from "vitest";
import { render } from "src/test-utils";
import { AddCircleIcon } from "./AddCircleIcon";

describe("AddCircleIcon", () => {
  it("renders correctly", () => {
    const { container, getByTestId } = render(<AddCircleIcon />);

    const element = getByTestId("add-circle-icon");

    expect(element).toBeInTheDocument();
    expect(element).toHaveAttribute("width");
    expect(element).toHaveAttribute("stroke");

    expect(container).toMatchSnapshot();
  });
});
