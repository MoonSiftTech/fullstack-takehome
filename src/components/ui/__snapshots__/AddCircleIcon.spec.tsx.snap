// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddCircleIcon > renders correctly 1`] = `
<div>
  <svg
    data-testid="add-circle-icon"
    fill="none"
    height="24"
    stroke="currentColor"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      r="10"
      stroke-width="1"
    />
    <line
      stroke-width="1"
      x1="12"
      x2="12"
      y1="8"
      y2="16"
    />
    <line
      stroke-width="1"
      x1="8"
      x2="16"
      y1="12"
      y2="12"
    />
  </svg>
</div>
`;
